## Practice Exercise 2

In this assignment, you will try to find some interesting insights into a few movies released between 1916 and 2016, using Python. You will have to download a movie dataset, write Python code to explore the data, gain insights into the movies, actors, directors, and collections, and submit the code.

#### Some tips before starting the assignment

1. Identify the task to be performed correctly, and only then proceed to write the required code. Don’t perform any incorrect analysis or look for information that isn’t required for the assignment.
2. In some cases, the variable names have already been assigned, and you just need to write code against them. In other cases, the names to be given are mentioned in the instructions. We strongly advise you to use the mentioned names only.
3. Always keep inspecting your data frame after you have performed a particular set of operations.
4. There are some checkpoints given in the IPython notebook provided. They're just useful pieces of information you can use to check if the result you have obtained after performing a particular task is correct or not.
5. Note that you will be asked to refer to documentation for solving some of the questions. That is done on purpose for you to learn new commands and also how to use the documentation.

In [1]:
# Import the numpy and pandas packages

import numpy as np
import pandas as pd

### Task 1: Reading and Inspection

**Subtask 1.1: Import and read**

Import and read the movie database. Store it in a variable called `movies`.

In [2]:
# Write your code for importing the csv file here
movies = pd.read_csv("movies.csv")
movies.head()

color      director_name  num_critic_for_reviews  duration  \
0  Color      James Cameron                   723.0     178.0   
1  Color     Gore Verbinski                   302.0     169.0   
2  Color         Sam Mendes                   602.0     148.0   
3  Color  Christopher Nolan                   813.0     164.0   
4  Color     Andrew Stanton                   462.0     132.0   

   director_facebook_likes  actor_3_facebook_likes      actor_2_name  \
0                      0.0                   855.0  Joel David Moore   
1                    563.0                  1000.0     Orlando Bloom   
2                      0.0                   161.0      Rory Kinnear   
3                  22000.0                 23000.0    Christian Bale   
4                    475.0                   530.0   Samantha Morton   

   actor_1_facebook_likes        gross                           genres  ...  \
0                  1000.0  760505847.0  Action|Adventure|Fantasy|Sci-Fi  ...   
1                 40000.0  309404152.0         Action|Adventure|Fantasy  ...   
2                 11000.0  200074175.0        Action|Adventure|Thriller  ...   
3                 27000.0  448130642.0                  Action|Thriller  ...   
4                   640.0   73058679.0          Action|Adventure|Sci-Fi  ...   

  num_user_for_reviews language  country  content_rating       budget  \
0               3054.0  English      USA           PG-13  237000000.0   
1               1238.0  English      USA           PG-13  300000000.0   
2                994.0  English       UK           PG-13  245000000.0   
3               2701.0  English      USA           PG-13  250000000.0   
4                738.0  English      USA           PG-13  263700000.0   

   title_year actor_2_facebook_likes imdb_score  aspect_ratio  \
0      2009.0                  936.0        7.9          1.78   
1      2007.0                 5000.0        7.1          2.35   
2      2015.0                  393.0        6.8          2.35   
3      2012.0                23000.0        8.5          2.35   
4      2012.0                  632.0        6.6          2.35   

  movie_facebook_likes  
0                33000  
1                    0  
2                85000  
3               164000  
4                24000  

[5 rows x 28 columns]

**Subtask 1.2: Inspect the dataframe**

Inspect the dataframe's columns, shapes, variable types etc.

In [3]:
# Write your code for inspection here
movies.shape

(3853, 28)

#### <font color='red'>Question 1: How many rows and columns are present in the dataframe? </font>
-  <font color='red'>(3821, 26)</font>
-  <font color='red'>(3879, 28)</font>
-  <font color='red'>(3853, 28)</font>
-  <font color='red'>(3866, 26)</font>

In [4]:
movies.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3853 entries, 0 to 3852
Data columns (total 28 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   color                      3851 non-null   object 
 1   director_name              3853 non-null   object 
 2   num_critic_for_reviews     3852 non-null   float64
 3   duration                   3852 non-null   float64
 4   director_facebook_likes    3853 non-null   float64
 5   actor_3_facebook_likes     3847 non-null   float64
 6   actor_2_name               3852 non-null   object 
 7   actor_1_facebook_likes     3853 non-null   float64
 8   gross                      3853 non-null   float64
 9   genres                     3853 non-null   object 
 10  actor_1_name               3853 non-null   object 
 11  movie_title                3853 non-null   object 
 12  num_voted_users            3853 non-null   int64  
 13  cast_total_facebook_likes  3853 non-null   int64

#### <font color='red'>Question 2: How many columns have null values present in them? Try writing a code for this instead of counting them manually.</font>

-  <font color='red'>3</font>
-  <font color='red'>6</font>
-  <font color='red'>9</font>
-  <font color='red'>12</font>

### Task 2: Cleaning the Data

**Subtask 2.1: Drop unecessary columns**

For this assignment, you will mostly be analyzing the movies with respect to the ratings, gross collection, popularity of movies, etc. So many of the columns in this dataframe are not required. So it is advised to drop the following columns.
-  color
-  director_facebook_likes
-  actor_1_facebook_likes
-  actor_2_facebook_likes
-  actor_3_facebook_likes
-  actor_2_name
-  cast_total_facebook_likes
-  actor_3_name
-  duration
-  facenumber_in_poster
-  content_rating
-  country
-  movie_imdb_link
-  aspect_ratio
-  plot_keywords

In [5]:
# Check the 'drop' function in the Pandas library - dataframe.drop(list_of_unnecessary_columns, axis = )
# Write your code for dropping the columns here. It is advised to keep inspecting the dataframe after each set of operations
movies = movies.drop(["color","director_facebook_likes","actor_1_facebook_likes","actor_2_facebook_likes","actor_3_facebook_likes","actor_2_name","cast_total_facebook_likes","actor_3_name","duration","facenumber_in_poster","content_rating","country","movie_imdb_link","aspect_ratio","plot_keywords"], axis =1)
movies.columns

Index(['director_name', 'num_critic_for_reviews', 'gross', 'genres',
       'actor_1_name', 'movie_title', 'num_voted_users',
       'num_user_for_reviews', 'language', 'budget', 'title_year',
       'imdb_score', 'movie_facebook_likes'],
      dtype='object')

#### <font color='red'>Question 3: What is the count of columns in the new dataframe? </font>
-  <font color='red'>10</font>
-  <font color='red'>13</font>
-  <font color='red'>15</font>
-  <font color='red'>17</font>

**Subtask 2.2: Inspect Null values**

As you have seen above, there are null values in multiple columns of the dataframe 'movies'. Find out the percentage of null values in each column of the dataframe 'movies'. 

In [6]:
# Write you code here
movies.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3853 entries, 0 to 3852
Data columns (total 13 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   director_name           3853 non-null   object 
 1   num_critic_for_reviews  3852 non-null   float64
 2   gross                   3853 non-null   float64
 3   genres                  3853 non-null   object 
 4   actor_1_name            3853 non-null   object 
 5   movie_title             3853 non-null   object 
 6   num_voted_users         3853 non-null   int64  
 7   num_user_for_reviews    3853 non-null   float64
 8   language                3850 non-null   object 
 9   budget                  3853 non-null   float64
 10  title_year              3853 non-null   float64
 11  imdb_score              3853 non-null   float64
 12  movie_facebook_likes    3853 non-null   int64  
dtypes: float64(6), int64(2), object(5)
memory usage: 391.4+ KB


#### <font color='red'>Question 4: Which column has the highest percentage of null values? </font>
-  <font color='red'>language</font>
-  <font color='red'>genres</font>
-  <font color='red'>num_critic_for_reviews</font>
-  <font color='red'>imdb_score</font>

**Subtask 2.3: Fill NaN values**

You might notice that the `language` column has some NaN values. Here, on inspection, you will see that it is safe to replace all the missing values with `'English'`.

In [7]:
# Write your code for filling the NaN values in the 'language' column here
movies[movies["language"] == 'English'].count()

director_name             3671
num_critic_for_reviews    3670
gross                     3671
genres                    3671
actor_1_name              3671
movie_title               3671
num_voted_users           3671
num_user_for_reviews      3671
language                  3671
budget                    3671
title_year                3671
imdb_score                3671
movie_facebook_likes      3671
dtype: int64

#### <font color='red'>Question 5: What is the count of movies made in English language after replacing the NaN values with English? </font>
-  <font color='red'>3670</font>
-  <font color='red'>3674</font>
-  <font color='red'>3668</font>
-  <font color='red'>3672</font>

### Task 3: Data Analysis

**Subtask 3.1: Change the unit of columns**

Convert the unit of the `budget` and `gross` columns from `$` to `million $`.

In [9]:
# Write your code for unit conversion here
movies["gross"]= movies["gross"]/1000000
movies["budget"]=movies["budget"]/1000000
movies.head()

director_name  num_critic_for_reviews     gross  \
0      James Cameron                   723.0  0.000761   
1     Gore Verbinski                   302.0  0.000309   
2         Sam Mendes                   602.0  0.000200   
3  Christopher Nolan                   813.0  0.000448   
4     Andrew Stanton                   462.0  0.000073   

                            genres     actor_1_name  \
0  Action|Adventure|Fantasy|Sci-Fi      CCH Pounder   
1         Action|Adventure|Fantasy      Johnny Depp   
2        Action|Adventure|Thriller  Christoph Waltz   
3                  Action|Thriller        Tom Hardy   
4          Action|Adventure|Sci-Fi     Daryl Sabara   

                                 movie_title  num_voted_users  \
0                                    Avatar            886204   
1  Pirates of the Caribbean: At World's End            471220   
2                                   Spectre            275868   
3                     The Dark Knight Rises           1144337   
4                               John Carter            212204   

   num_user_for_reviews language    budget  title_year  imdb_score  \
0                3054.0  English  0.000237      2009.0         7.9   
1                1238.0  English  0.000300      2007.0         7.1   
2                 994.0  English  0.000245      2015.0         6.8   
3                2701.0  English  0.000250      2012.0         8.5   
4                 738.0  English  0.000264      2012.0         6.6   

   movie_facebook_likes  
0                 33000  
1                     0  
2                 85000  
3                164000  
4                 24000

**Subtask 3.2: Find the movies with highest profit**

   1. Create a new column called `profit` which contains the difference of the two columns: `gross` and `budget`.
   2. Sort the dataframe using the `profit` column as reference. (Find which command can be used here to sort entries from the documentation)
   3. Extract the top ten profiting movies in descending order and store them in a new dataframe - `top10`

In [10]:
# Write your code for creating the profit column here
movies["profit"]=movies["gross"]-movies["budget"]

In [11]:
# Write your code for sorting the dataframe here
movies.sort_values("profit",ascending = False ).head(10)

director_name  num_critic_for_reviews     gross  \
0         James Cameron                   723.0  0.000761   
28      Colin Trevorrow                   644.0  0.000652   
25        James Cameron                   315.0  0.000659   
2704       George Lucas                   282.0  0.000461   
2748   Steven Spielberg                   215.0  0.000435   
16          Joss Whedon                   703.0  0.000623   
482        Roger Allers                   186.0  0.000423   
230        George Lucas                   320.0  0.000475   
64    Christopher Nolan                   645.0  0.000533   
419           Gary Ross                   673.0  0.000408   

                                        genres         actor_1_name  \
0              Action|Adventure|Fantasy|Sci-Fi          CCH Pounder   
28            Action|Adventure|Sci-Fi|Thriller  Bryce Dallas Howard   
25                               Drama|Romance    Leonardo DiCaprio   
2704           Action|Adventure|Fantasy|Sci-Fi        Harrison Ford   
2748                             Family|Sci-Fi         Henry Thomas   
16                     Action|Adventure|Sci-Fi      Chris Hemsworth   
482   Adventure|Animation|Drama|Family|Musical    Matthew Broderick   
230            Action|Adventure|Fantasy|Sci-Fi      Natalie Portman   
64                 Action|Crime|Drama|Thriller       Christian Bale   
419            Adventure|Drama|Sci-Fi|Thriller    Jennifer Lawrence   

                                     movie_title  num_voted_users  \
0                                        Avatar            886204   
28                               Jurassic World            418214   
25                                      Titanic            793059   
2704         Star Wars: Episode IV - A New Hope            911097   
2748                 E.T. the Extra-Terrestrial            281842   
16                                 The Avengers            995415   
482                               The Lion King            644348   
230   Star Wars: Episode I - The Phantom Menace            534658   
64                              The Dark Knight           1676169   
419                            The Hunger Games            701607   

      num_user_for_reviews language    budget  title_year  imdb_score  \
0                   3054.0  English  0.000237      2009.0         7.9   
28                  1290.0  English  0.000150      2015.0         7.0   
25                  2528.0  English  0.000200      1997.0         7.7   
2704                1470.0  English  0.000011      1977.0         8.7   
2748                 515.0  English  0.000010      1982.0         7.9   
16                  1722.0  English  0.000220      2012.0         8.1   
482                  656.0  English  0.000045      1994.0         8.5   
230                 3597.0  English  0.000115      1999.0         6.5   
64                  4667.0  English  0.000185      2008.0         9.0   
419                 1959.0  English  0.000078      2012.0         7.3   

      movie_facebook_likes    profit  
0                    33000  0.000524  
28                  150000  0.000502  
25                   26000  0.000459  
2704                 33000  0.000450  
2748                 34000  0.000424  
16                  123000  0.000403  
482                  17000  0.000378  
230                  13000  0.000360  
64                   37000  0.000348  
419                 140000  0.000330

In [12]:
top10 = movies.sort_values("profit",ascending = False ).head(10)# Write your code to get the top 10 profiting movies here

**Checkpoint:** You might spot two movies directed by `James Cameron` in the list.

#### <font color='red'>Question 6: Which movie is ranked 5th from the top in the list obtained? </font>
-  <font color='red'>E.T. the Extra-Terrestrial</font>
-  <font color='red'>The Avengers</font>
-  <font color='red'>The Dark Knight</font>
-  <font color='red'>Titanic</font>

**Subtask 3.3: Find IMDb Top 250**

Create a new dataframe `IMDb_Top_250` and store the top 250 movies with the highest IMDb Rating (corresponding to the column: `imdb_score`). Also make sure that for all of these movies, the `num_voted_users` is greater than 25,000. 

Also add a `Rank` column containing the values 1 to 250 indicating the ranks of the corresponding films.

In [13]:
# Write your code for extracting the top 250 movies as per the IMDb score here. Make sure that you store it in a new dataframe 
# and name that dataframe as 'IMDb_Top_250'
IMDb_Top_250 = movies[(movies["imdb_score"]>=7.5)&(movies["num_voted_users"]>25000)]
IMDb_Top_250 = IMDb_Top_250.iloc[0:250,:]
IMDb_Top_250["Rank"]= range(1,251)
IMDb_Top_250.head()

director_name  num_critic_for_reviews     gross  \
0      James Cameron                   723.0  0.000761   
3  Christopher Nolan                   813.0  0.000448   
6       Nathan Greno                   324.0  0.000201   
7        Joss Whedon                   635.0  0.000459   
8        David Yates                   375.0  0.000302   

                                              genres     actor_1_name  \
0                    Action|Adventure|Fantasy|Sci-Fi      CCH Pounder   
3                                    Action|Thriller        Tom Hardy   
6  Adventure|Animation|Comedy|Family|Fantasy|Musi...     Brad Garrett   
7                            Action|Adventure|Sci-Fi  Chris Hemsworth   
8                   Adventure|Family|Fantasy|Mystery     Alan Rickman   

                               movie_title  num_voted_users  \
0                                  Avatar            886204   
3                   The Dark Knight Rises           1144337   
6                                 Tangled            294810   
7                 Avengers: Age of Ultron            462669   
8  Harry Potter and the Half-Blood Prince            321795   

   num_user_for_reviews language    budget  title_year  imdb_score  \
0                3054.0  English  0.000237      2009.0         7.9   
3                2701.0  English  0.000250      2012.0         8.5   
6                 387.0  English  0.000260      2010.0         7.8   
7                1117.0  English  0.000250      2015.0         7.5   
8                 973.0  English  0.000250      2009.0         7.5   

   movie_facebook_likes    profit  Rank  
0                 33000  0.000524     1  
3                164000  0.000198     2  
6                 29000 -0.000059     3  
7                118000  0.000209     4  
8                 10000  0.000052     5

#### <font color='red'>Question 7: Suppose movies are divided into 5 buckets based on the IMDb ratings: </font>
-  <font color='red'>7.5 to 8</font>
-  <font color='red'>8 to 8.5</font>
-  <font color='red'>8.5 to 9</font>
-  <font color='red'>9 to 9.5</font>
-  <font color='red'>9.5 to 10</font>

<font color = 'red'> Which bucket holds the maximum number of movies from *IMDb_Top_250*? </font>

In [14]:
IMDb_Top_250[(IMDb_Top_250["imdb_score"]>=7.5)& (IMDb_Top_250["imdb_score"]<8)].value_counts().count()

166

In [15]:
IMDb_Top_250[(IMDb_Top_250["imdb_score"]>=8)& (IMDb_Top_250["imdb_score"]<8.5)].value_counts().count()

63

In [16]:
IMDb_Top_250[(IMDb_Top_250["imdb_score"]>=8.5)& (IMDb_Top_250["imdb_score"]<9)].value_counts().count()

20

In [17]:
IMDb_Top_250[(IMDb_Top_250["imdb_score"]>=9)& (IMDb_Top_250["imdb_score"]<9.5)].value_counts().count()

1

In [18]:
IMDb_Top_250[(IMDb_Top_250["imdb_score"]>=9.5)& (IMDb_Top_250["imdb_score"]<10)].value_counts().count()

0

**Subtask 3.4: Find the critic-favorite and audience-favorite actors**

   1. Create three new dataframes namely, `Meryl_Streep`, `Leo_Caprio`, and `Brad_Pitt` which contain the movies in which the actors: 'Meryl Streep', 'Leonardo DiCaprio', and 'Brad Pitt' are the lead actors. Use only the `actor_1_name` column for extraction. Also, make sure that you use the names 'Meryl Streep', 'Leonardo DiCaprio', and 'Brad Pitt' for the said extraction.
   2. Append the rows of all these dataframes and store them in a new dataframe named `Combined`.
   3. Group the combined dataframe using the `actor_1_name` column.
   4. Find the mean of the `num_critic_for_reviews` and `num_user_for_review` and identify the actors which have the highest mean.

In [19]:
# Write your code for creating three new dataframes here
Meryl_Streep = movies[movies["actor_1_name"]=="Meryl Streep"] 
Meryl_Streep# Include all movies in which Meryl_Streep is the lead

director_name  num_critic_for_reviews     gross  \
392     Nancy Meyers                   187.0  0.000113   
1038   Curtis Hanson                    42.0  0.000047   
1132     Nora Ephron                   252.0  0.000094   
1322   David Frankel                   208.0  0.000125   
1390  Robert Redford                   227.0  0.000015   
1471  Sydney Pollack                    66.0  0.000087   
1514   David Frankel                   234.0  0.000064   
1563   Carl Franklin                    64.0  0.000023   
1784  Stephen Daldry                   174.0  0.000042   
2500  Phyllida Lloyd                   331.0  0.000030   
2793   Robert Altman                   211.0  0.000020   

                               genres  actor_1_name  \
392              Comedy|Drama|Romance  Meryl Streep   
1038  Action|Adventure|Crime|Thriller  Meryl Streep   
1132          Biography|Drama|Romance  Meryl Streep   
1322             Comedy|Drama|Romance  Meryl Streep   
1390               Drama|Thriller|War  Meryl Streep   
1471          Biography|Drama|Romance  Meryl Streep   
1514             Comedy|Drama|Romance  Meryl Streep   
1563                            Drama  Meryl Streep   
1784                    Drama|Romance  Meryl Streep   
2500          Biography|Drama|History  Meryl Streep   
2793               Comedy|Drama|Music  Meryl Streep   

                    movie_title  num_voted_users  num_user_for_reviews  \
392           It's Complicated             69860                 214.0   
1038            The River Wild             32544                  69.0   
1132             Julie & Julia             79264                 277.0   
1322     The Devil Wears Prada            286178                 631.0   
1390           Lions for Lambs             41170                 298.0   
1471             Out of Africa             52339                 200.0   
1514              Hope Springs             34258                 178.0   
1563            One True Thing              9283                 112.0   
1784                 The Hours            102123                 660.0   
2500             The Iron Lady             82327                 350.0   
2793  A Prairie Home Companion             19655                 280.0   

     language    budget  title_year  imdb_score  movie_facebook_likes  \
392   English  0.000085      2009.0         6.6                     0   
1038  English  0.000045      1994.0         6.3                     0   
1132  English  0.000040      2009.0         7.0                 13000   
1322  English  0.000035      2006.0         6.8                     0   
1390  English  0.000035      2007.0         6.2                     0   
1471  English  0.000031      1985.0         7.2                     0   
1514  English  0.000030      2012.0         6.3                     0   
1563  English  0.000030      1998.0         7.0                   592   
1784  English  0.000025      2002.0         7.6                     0   
2500  English  0.000013      2011.0         6.4                 18000   
2793  English  0.000010      2006.0         6.8                   683   

        profit  
392   0.000028  
1038  0.000002  
1132  0.000054  
1322  0.000090  
1390 -0.000020  
1471  0.000056  
1514  0.000034  
1563 -0.000007  
1784  0.000017  
2500  0.000017  
2793  0.000010

In [20]:
Leo_Caprio = movies[movies["actor_1_name"]=="Leonardo DiCaprio"]
Leo_Caprio# Include all movies in which Leo_Caprio is the lead

director_name  num_critic_for_reviews     gross  \
25            James Cameron                   315.0  0.000659   
49             Baz Luhrmann                   490.0  0.000145   
94        Christopher Nolan                   642.0  0.000293   
173   Alejandro G. Iñárritu                   556.0  0.000184   
246         Martin Scorsese                   267.0  0.000103   
283       Quentin Tarantino                   765.0  0.000163   
293            Edward Zwick                   166.0  0.000057   
294         Martin Scorsese                   606.0  0.000117   
312         Martin Scorsese                   233.0  0.000078   
347         Martin Scorsese                   352.0  0.000132   
431         Martin Scorsese                   490.0  0.000128   
608            Ridley Scott                   238.0  0.000039   
859        Steven Spielberg                   194.0  0.000164   
934             Danny Boyle                   118.0  0.000040   
1046             Sam Mendes                   323.0  0.000023   
1336        Randall Wallace                    83.0  0.000057   
1364         Clint Eastwood                   392.0  0.000037   
1457              Sam Raimi                    63.0  0.000019   
1909             Jerry Zaks                    45.0  0.000013   
2484           Baz Luhrmann                   106.0  0.000046   
3025           Baz Luhrmann                   490.0  0.000145   

                                genres       actor_1_name  \
25                       Drama|Romance  Leonardo DiCaprio   
49                       Drama|Romance  Leonardo DiCaprio   
94    Action|Adventure|Sci-Fi|Thriller  Leonardo DiCaprio   
173   Adventure|Drama|Thriller|Western  Leonardo DiCaprio   
246                    Biography|Drama  Leonardo DiCaprio   
283                      Drama|Western  Leonardo DiCaprio   
293           Adventure|Drama|Thriller  Leonardo DiCaprio   
294       Biography|Comedy|Crime|Drama  Leonardo DiCaprio   
312                        Crime|Drama  Leonardo DiCaprio   
347               Crime|Drama|Thriller  Leonardo DiCaprio   
431                   Mystery|Thriller  Leonardo DiCaprio   
608              Action|Drama|Thriller  Leonardo DiCaprio   
859              Biography|Crime|Drama  Leonardo DiCaprio   
934           Adventure|Drama|Thriller  Leonardo DiCaprio   
1046                     Drama|Romance  Leonardo DiCaprio   
1336                  Action|Adventure  Leonardo DiCaprio   
1364             Biography|Crime|Drama  Leonardo DiCaprio   
1457           Action|Thriller|Western  Leonardo DiCaprio   
1909                             Drama  Leonardo DiCaprio   
2484                     Drama|Romance  Leonardo DiCaprio   
3025                     Drama|Romance  Leonardo DiCaprio   

                    movie_title  num_voted_users  num_user_for_reviews  \
25                     Titanic            793059                2528.0   
49            The Great Gatsby            362912                 753.0   
94                   Inception           1468200                2803.0   
173               The Revenant            406020                1188.0   
246                The Aviator            264318                 799.0   
283           Django Unchained            955174                1193.0   
293              Blood Diamond            400292                 657.0   
294    The Wolf of Wall Street            780588                1138.0   
312          Gangs of New York            314033                1166.0   
347               The Departed            873649                2054.0   
431             Shutter Island            786092                 964.0   
608               Body of Lies            174248                 263.0   
859        Catch Me If You Can            525801                 667.0   
934                  The Beach            176169                 548.0   
1046        Revolutionary Road            152591                 414.0   
1336  The Man in the Iron Mask            125219        

In [21]:
Brad_Pitt = movies[movies["actor_1_name"]=="Brad Pitt"]
Brad_Pitt# Include all movies in which Brad_Pitt is the lead

director_name  num_critic_for_reviews         gross  \
97            David Fincher                   362.0  1.274908e-04   
142       Wolfgang Petersen                   220.0  1.332283e-04   
243       Steven Soderbergh                   198.0  1.255316e-04   
244              Doug Liman                   233.0  1.863361e-04   
367              Tony Scott                   142.0  2.687100e-08   
383       Steven Soderbergh                   186.0  1.834058e-04   
448              David Ayer                   406.0  8.570712e-05   
579     Jean-Jacques Annaud                    76.0  3.790151e-05   
646           David Fincher                   315.0  3.702340e-05   
749         Patrick Gilmore                    98.0  2.628832e-05   
887             Neil Jordan                   120.0  1.052646e-04   
1397        Terrence Malick                   584.0  1.330332e-05   
1608         Andrew Dominik                   273.0  3.904982e-06   
2030  Alejandro G. Iñárritu                   285.0  3.430077e-05   
2146    Angelina Jolie Pitt                   131.0  5.310090e-07   
2428         Andrew Dominik                   414.0  1.493857e-05   
2602             Tony Scott                   122.0  1.228150e-05   

                                                 genres actor_1_name  \
97                                Drama|Fantasy|Romance    Brad Pitt   
142                                           Adventure    Brad Pitt   
243                                      Crime|Thriller    Brad Pitt   
244                Action|Comedy|Crime|Romance|Thriller    Brad Pitt   
367                               Action|Crime|Thriller    Brad Pitt   
383                                      Crime|Thriller    Brad Pitt   
448                                    Action|Drama|War    Brad Pitt   
579               Adventure|Biography|Drama|History|War    Brad Pitt   
646                                               Drama    Brad Pitt   
749   Adventure|Animation|Comedy|Drama|Family|Fantas...    Brad Pitt   
887                                Drama|Fantasy|Horror    Brad Pitt   
1397                                      Drama|Fantasy    Brad Pitt   
1608              Biography|Crime|Drama|History|Western    Brad Pitt   
2030                                              Drama    Brad Pitt   
2146                                      Drama|Romance    Brad Pitt   
2428                                     Crime|Thriller    Brad Pitt   
2602                Action|Crime|Drama|Romance|Thriller    Brad Pitt   

                                            movie_title  num_voted_users  \
97                 The Curious Case of Benjamin Button            459346   
142                                               Troy            381672   
243                                     Ocean's Twelve            284852   
244                                   Mr. & Mrs. Smith            348861   
367                                           Spy Game            121259   
383                                     Ocean's Eleven            402645   
448                                               Fury            303185   
579                               Seven Years in Tibet             96385   
646                                         Fight Club           1347461   
749                   Sinbad: Legend of the Seven Seas             36144   
887   Interview with the Vampire: The Vampire Chroni...           239752   
1397                                  The Tree of Life            136367   
1608  The Assassination of Jesse James by the Coward...           136104   
2030                                             Babel            243799   
2146                                        By the Sea              7976   
2428                               Killing Them Softly            111625   
2602                                      True Romance            163492   

      num_user_for_reviews language    budget  title_year  imdb_score  \
97                   822.0  Engl

In [25]:
# Write your code for combining the three dataframes here
Combined = pd.concat([Meryl_Streep,Leo_Caprio,Brad_Pitt], axis = 0)
Combined

director_name  num_critic_for_reviews         gross  \
392            Nancy Meyers                   187.0  1.127035e-04   
1038          Curtis Hanson                    42.0  4.681575e-05   
1132            Nora Ephron                   252.0  9.412543e-05   
1322          David Frankel                   208.0  1.247330e-04   
1390         Robert Redford                   227.0  1.499807e-05   
1471         Sydney Pollack                    66.0  8.710000e-05   
1514          David Frankel                   234.0  6.353601e-05   
1563          Carl Franklin                    64.0  2.320944e-05   
1784         Stephen Daldry                   174.0  4.159783e-05   
2500         Phyllida Lloyd                   331.0  2.995944e-05   
2793          Robert Altman                   211.0  2.033861e-05   
25            James Cameron                   315.0  6.586723e-04   
49             Baz Luhrmann                   490.0  1.448128e-04   
94        Christopher Nolan                   642.0  2.925689e-04   
173   Alejandro G. Iñárritu                   556.0  1.836359e-04   
246         Martin Scorsese                   267.0  1.026088e-04   
283       Quentin Tarantino                   765.0  1.628046e-04   
293            Edward Zwick                   166.0  5.736626e-05   
294         Martin Scorsese                   606.0  1.168667e-04   
312         Martin Scorsese                   233.0  7.767964e-05   
347         Martin Scorsese                   352.0  1.323734e-04   
431         Martin Scorsese                   490.0  1.279684e-04   
608            Ridley Scott                   238.0  3.938044e-05   
859        Steven Spielberg                   194.0  1.644352e-04   
934             Danny Boyle                   118.0  3.977860e-05   
1046             Sam Mendes                   323.0  2.287781e-05   
1336        Randall Wallace                    83.0  5.687637e-05   
1364         Clint Eastwood                   392.0  3.730495e-05   
1457              Sam Raimi                    63.0  1.863654e-05   
1909             Jerry Zaks                    45.0  1.278251e-05   
2484           Baz Luhrmann                   106.0  4.633873e-05   
3025           Baz Luhrmann                   490.0  1.448128e-04   
97            David Fincher                   362.0  1.274908e-04   
142       Wolfgang Petersen                   220.0  1.332283e-04   
243       Steven Soderbergh                   198.0  1.255316e-04   
244              Doug Liman                   233.0  1.863361e-04   
367              Tony Scott                   142.0  2.687100e-08   
383       Steven Soderbergh                   186.0  1.834058e-04   
448              David Ayer                   406.0  8.570712e-05   
579     Jean-Jacques Annaud                    76.0  3.790151e-05   
646           David Fincher                   315.0  3.702340e-05   
749         Patrick Gilmore                    98.0  2.628832e-05   
887             Neil Jordan                   120.0  1.052646e-04   
1397        Terrence Malick                   584.0  1.330332e-05   
1608         Andrew Dominik                   273.0  3.904982e-06   
2030  Alejandro G. Iñárritu                   285.0  3.430077e-05   
2146    Angelina Jolie Pitt                   131.0  5.310090e-07   
2428         Andrew Dominik                   414.0  1.493857e-05   
2602             Tony Scott                   122.0  1.228150e-05   

                                                 genres       actor_1_name  \
392                                Comedy|Drama|Romance       Meryl Streep   
1038                    Action|Adventure|Crime|Thriller       Meryl Streep   
1132                            Biography|Drama|Romance       Meryl Streep   
1322                               Comedy|Drama|Romance       Meryl Streep   
1390                                 Drama|Thriller|War       Meryl Streep   
1471                            Biography|Drama|Romance       Meryl Streep   
1514             

In [26]:
# Write your code for grouping the combined dataframe here
Combined = Combined.groupby(by =["actor_1_name"]).mean()
Combined

num_critic_for_reviews     gross  num_voted_users  \
actor_1_name                                                           
Brad Pitt                      245.000000  0.000066    283583.823529   
Leonardo DiCaprio              330.190476  0.000126    441958.857143   
Meryl Streep                   181.454545  0.000060     73545.545455   

                   num_user_for_reviews    budget   title_year  imdb_score  \
actor_1_name                                                                 
Brad Pitt                    742.352941  0.000069  2004.352941    7.152941   
Leonardo DiCaprio            914.476190  0.000082  2005.476190    7.495238   
Meryl Streep                 297.181818  0.000034  2003.545455    6.745455   

                   movie_facebook_likes    profit  
actor_1_name                                       
Brad Pitt                  14051.764706 -0.000003  
Leonardo DiCaprio          52190.476190  0.000043  
Meryl Streep                2934.090909  0.000025

In [ ]:
# Write the code for finding  of critic reviews and the mean audience reviews here


#### <font color='red'>Question 8: Which actor is highest rated among the three actors according to the user reviews? </font>
-  <font color='red'>Meryl Streep</font>
-  <font color='red'>Leonardo DiCaprio</font>
-  <font color='red'>Brad Pitt</font>

In [ ]:
# Ans: Leonardo Dicaprio

#### <font color='red'>Question 9: Which actor is highest rated among the three actors according to the critics?</font>
-  <font color='red'>Meryl Streep</font>
-  <font color='red'>Leonardo DiCaprio</font>
-  <font color='red'>Brad Pitt</font>

In [ ]:
# Ans: Leonardo Dicaprio